cmd/trader/trader.go

```diff
@@ -34,146 +41,232 @@ func main() {
 	// Parse the option parameters
 	numberOfOrders := flag.Int("order", 10, "number of orders")
 	timeInterval := flag.Int("time", 15, "time interval in second")
+	cancelRequest := flag.Bool("cancel", false, "cancel request")
 
-	// Get nodes/darkPool details
-	multiAddresses := getNodesDetails()
-	nodes := make([]identity.MultiAddress, len(multiAddresses))
-	for i := 0; i < len(multiAddresses); i++ {
-		multi, err := identity.NewMultiAddressFromString(multiAddresses[i])
-		if err != nil {
-			log.Fatal(err)
-		}
-		nodes[i] = multi
-		log.Println(base58.Encode(nodes[i].ID()))
-	}
+	// Logger
+	logs, err := logger.NewLogger(logger.Options{
+		Plugins: []logger.PluginOptions{
+			logger.PluginOptions{
+				File: &logger.FilePluginOptions{
+					Path: "stdout",
+				},
+			},
+		}})
 
-	// Create a trader address
+	// Create a test trader address
 	address, _, err := identity.NewAddress()
 	if err != nil {
-		log.Fatal(err)
+		panic(err)
 	}
 	multi, err := identity.NewMultiAddressFromString("/ip4/0.0.0.0/tcp/80/republic/" + address.String())
 	if err != nil {
-		log.Fatal(err)
+		panic(err)
 	}
 	log.Println("Trader Address: ", address)
 
-	// Keep sending order fragment
-	for {
-		// Get orders details from Binance
-		resp, err := http.Get(fmt.Sprintf("https://api.binance.com/api/v1/depth?symbol=ETHBTC&limit=%v", *numberOfOrders))
-		if err != nil {
-			log.Fatal("fail to get data from binance")
-		}
+	// Get the Dark Node Registry
+	registrar, err := getDarkNodeRegistrar()
+	if err != nil {
+		panic(err)
+	}
 
-		response, err := ioutil.ReadAll(resp.Body)
-		orderBook := new(OrderBook)
-		err = json.Unmarshal(response, orderBook)
+	// Get the Dark Ocean
+	ocean, err := dark.NewOcean(logs, 5, registrar)
 
-		// Generate order from the Binance data
-		sellOrders := make([]*order.Order, len(orderBook.Asks))
-		for i, j := range orderBook.Asks {
-			price, err := strconv.ParseFloat(j[0], 10)
-			price = price * 1000000000000
+	// Get the Dark Pools
+	pools := ocean.GetPools()
+
+	if *cancelRequest {
+		cancelOrder(pools, multi)
+	} else {
+		// Keep sending order fragment
+		for {
+			// Get orders details from Binance
+			resp, err := http.Get(fmt.Sprintf("https://api.binance.com/api/v1/depth?symbol=ETHBTC&limit=%v", *numberOfOrders))
 			if err != nil {
-				log.Fatal("fail to parse the price into a big int")
+				log.Fatal("fail to get data from binance")
 			}
 
-			amount, err := strconv.ParseFloat(j[1], 10)
-			amount = amount * 1000000000000
-			if err != nil {
-				log.Fatal("fail to parse the amount into a big int")
+			response, err := ioutil.ReadAll(resp.Body)
+			orderBook := new(OrderBook)
+			err = json.Unmarshal(response, orderBook)
+
+			// Generate order from the Binance data
+			sellOrders := make([]*order.Order, len(orderBook.Asks))
+			for i, j := range orderBook.Asks {
+				price, err := strconv.ParseFloat(j[0], 10)
```

Comment:
Always check errors before doing anything. If the Parse fails, the multiplication is meaningless.